File: tasks/android-java/ToDo/app/src/main/java/com/dittolive/todo/NewTaskDialogFragment.java
package com.dittolive.todo;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import androidx.fragment.app.DialogFragment;

public class NewTaskDialogFragment extends DialogFragment {

    interface NewTaskDialogListener {
        void onDialogSave(DialogFragment dialogFragment, String task);
        void onDialogCancel(DialogFragment dialogFragment);
    }

    NewTaskDialogListener newTaskDialogListener = null;

    static NewTaskDialogFragment newInstance(int title) {
        NewTaskDialogFragment f = new NewTaskDialogFragment();
        Bundle args = new Bundle();
        args.putInt("dialog_title", title);
        f.setArguments(args);
        return f;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        int title = getArguments().getInt("dialog_title");
        final NewTaskDialogListener dialogListener = this.newTaskDialogListener;

        View dialogView = getActivity().getLayoutInflater().inflate(R.layout.dialog_new_task, null);
        final TextView task = dialogView.findViewById(R.id.editText);
        final DialogFragment dialogFragment = this;

        return new AlertDialog.Builder(getActivity())
            .setView(dialogView)
            .setTitle(title)
            .setPositiveButton(R.string.save,
                new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int whichButton) {
                        dialogListener.onDialogSave(dialogFragment, task.getText().toString());
                    }
                }
            )
            .setNegativeButton(android.R.string.cancel,
                new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int whichButton) {
                        dialogListener.onDialogCancel(dialogFragment);
                    }
                }
            )
            .create();
    }

    @SuppressWarnings("deprecation")
    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        try {
            this.newTaskDialogListener = (NewTaskDialogListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException("Activity must implement NewTaskDialogListener");
        }
    }

}

File: tasks/android-kotlin/ToDo/app/src/main/java/com/dittolive/todo/NewTaskDialogFragment.kt
package com.dittolive.todo

import android.app.Activity
import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import live.dittolive.todo.R

class NewTaskDialogFragment: DialogFragment() {



    interface NewTaskDialogListener {
        fun onDialogSave(dialog: DialogFragment, task: String)
        fun onDialogCancel(dialog: DialogFragment)
    }

    var newTaskDialogListener: NewTaskDialogListener? = null

    companion object {
        fun newInstance(): NewTaskDialogFragment {
            val newTaskDialogFragment = NewTaskDialogFragment()
            val args = Bundle()
            newTaskDialogFragment.arguments = args
            return newTaskDialogFragment
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog { // 5
        val title = "Add New Task"
        val builder = AlertDialog.Builder(activity)
        builder.setTitle(title)

        val dialogView = activity?.layoutInflater?.inflate(R.layout.dialog_new_task, null)
        val task = dialogView?.findViewById<TextView>(R.id.editText)

        builder.setView(dialogView)
            .setPositiveButton(R.string.save) { _, _ -> newTaskDialogListener?.onDialogSave(this, task?.text.toString()) }
            .setNegativeButton(android.R.string.cancel) { _, _ -> newTaskDialogListener?.onDialogCancel(this) }
        return builder.create()
    }

    @Deprecated("Deprecated in Java")
    @Suppress("DEPRECATION")
    override fun onAttach(activity: Activity) { // 6
        super.onAttach(activity)
        try {
            newTaskDialogListener = activity as NewTaskDialogListener
        } catch (e: ClassCastException) {
            throw ClassCastException("$activity must implement NewTaskDialogListener")
        }
    }
}
